File: Src/java/engine/src/main/java/org/opencds/cqf/cql/engine/elm/visiting/RepeatEvaluator.java
package org.opencds.cqf.cql.engine.elm.visiting;

import org.apache.commons.lang3.NotImplementedException;
import org.opencds.cqf.cql.engine.execution.Context;
import org.opencds.cqf.cql.engine.execution.State;

public class RepeatEvaluator {

    public static Object repeat(Object source, Object element, String scope) {
        // TODO
        throw new NotImplementedException("Repeat operation not yet implemented");
    }

    public static Object internalEvaluate(Object source, Object element, String scope, State state) {
        return repeat(source, element, scope);
    }
}

File: Src/java/engine/src/main/kotlin/org/opencds/cqf/cql/engine/elm/executing/RepeatEvaluator.kt
package org.opencds.cqf.cql.engine.elm.executing

import org.apache.commons.lang3.NotImplementedException
import org.opencds.cqf.cql.engine.execution.State

object RepeatEvaluator {
    fun repeat(source: Any?, element: Any?, scope: String?): Any {
        // TODO
        throw NotImplementedException("Repeat operation not yet implemented")
    }

    @JvmStatic
    fun internalEvaluate(source: Any?, element: Any?, scope: String?, state: State?): Any {
        return repeat(source, element, scope)
    }
}
